src/nvim/message.c

```diff
@@ -2262,7 +2267,7 @@ bool msg_use_msgsep(void)
 {
   // the full-screen scroll behavior doesn't really make sense with
   // 'ext_multigrid'
-  return ((dy_flags & DY_MSGSEP) || ui_has(kUIMultigrid));
+  return (((dy_flags & DY_MSGSEP) && p_ch > 0) || ui_has(kUIMultigrid));
```

Comment:
this change must be reverted. if the behavior is not correct with msgsep, it will break with multigrid as well, so it needs to be made correct.